### PR TITLE
Add elasticdl clean command to clean up local docker images

### LINKED ISSUE #1202: Add cleanup command to clean local built docker images
Currently, each ElasticDL job will build a local docker image, causing lots of images after submitting some jobs. Better to provide a command `elasticdl clean` to clean up some local docker images.
- `elasticdl clean --all` will clean all local docker images;
- `elasticdl clean --prefix ...` will only clean docker images with the given prefix. Here `prefix` refers to repository.

### PR DESCRIPTION
**Pull request by ywskycn**
To resolve #1202. Add `elasticdl clean` command to clean up local images. Support to different configs:
- `elasticdl clean --docker_image_repository repo1` is to clean all images with repository equals to `repo1`.
- `elasticdl clean --all` is to clean all local docker images.

### CODE REVIEW

File: elasticdl/python/elasticdl/api.py [hunk 1]
```diff
@@ -103,6 +104,28 @@ def predict(args):
...
+            "--docker_image_repository and --all cannot "
+            "be specified at the same time"
+        )
+    if not (args.docker_image_repository or args.all):
+        raise ValueError(
+            "Either --docker_image_repository or --all "
+            "needs to be configured"
+        )
+    repository = (
+        args.docker_image_repository if args.docker_image_repository else ""
```
Comment (terrytangyuan) [inline elasticdl/python/elasticdl/api.py]: `docker_image_repository` already defaults to `""` so this line doesn't seem necessary?

File: elasticdl/python/elasticdl/image_builder.py [hunk 2]
```diff
@@ -81,6 +79,55 @@ def build_and_push_docker_image(
     return image_name
 
 
+def remove_images(
```
Comment (terrytangyuan) [inline elasticdl/python/elasticdl/image_builder.py:82]: Is there an option `-f` for force removal?

File: elasticdl/python/elasticdl/image_builder.py [hunk 3]
```diff
@@ -81,6 +79,55 @@ def build_and_push_docker_image(
...
+    for image in images:
+        image_id = image.get("Id")
+        if image_id:
+            logger.info("Removing image %s" % image_id)
+            try:
+                client.remove_image(image_id)
+            except docker.errors.APIError as e:
+                logger.warning("Failed to delete image %s: %s" % (image_id, e))
+    # If removing all images, we also run prune to remove untagged images
+    if not docker_image_repository:
```
Comment (terrytangyuan) [inline elasticdl/python/elasticdl/image_builder.py:123]: Also check `--all`?

File: elasticdl/python/elasticdl/api.py (hunk 1 above)
Comment (ywskycn) [reply elasticdl/python/elasticdl/api.py]: Good catch. Yes, we can directly use `docker_image_repository` here. I introduced `--all` here to make sure users really want to clean up all images. `docker_image_repository` = ""  also means `--all`.

File: elasticdl/python/elasticdl/image_builder.py (hunk 3 above)
Comment (ywskycn) [reply elasticdl/python/elasticdl/image_builder.py:123]: `docker_image_repository` = "" actually means removing `all`, in the docker python API.

File: elasticdl/python/elasticdl/image_builder.py (hunk 2 above)
Comment (ywskycn) [reply elasticdl/python/elasticdl/image_builder.py:82]: `force` option available in docker python API for cases where you want to remove an image in a force way, without considering dependencies and containers. I put `force`=False here by default.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**